classes/Admin/Stats/Controller.php

```diff
@@ -0,0 +1,745 @@
+	/**
+	 * Array of actions.
+	 *
+	 * @var array
+	 */
+	private static $actions = [
+		'imagify_count_attachments_as',
+		'imagify_count_error_attachments_as',
+		'imagify_count_optimized_attachments_as',
+		'imagify_count_saving_data_as',
+		'imagify_calculate_total_size_images_library_as',
+		'imagify_calculate_average_size_images_per_month_as',
+	];
+
+	/**
+	 * Stats option name.
+	 *
+	 * @var string
+	 */
+	protected $stats_option = 'imagify_admin_stats';
+
+	/**
+	 * Admin stats option.
+	 *
+	 * @var array
+	 */
+	protected $admin_stats = [
+		'attachments_count' => 0,
+		'attachments_optimized_count' => 0,
+		'attachments_error_count' => 0,
+		'saving_data_count' => [
+			'count'          => 0,
+			'original_size'  => 0,
+			'optimized_size' => 0,
+			'percent' => 0,
+		],
+		'images_library_total_size' => 0,
+		'images_average_size_per_month' => 0,
+	];
+
+	/**
+	 * AS Queue group.
+	 *
+	 * @var string
+	 */
+	private $group = 'imagify-stats';
+
+	/**
+	 * Register actions to run in background.
+	 *
+	 * @return void
+	 */
+	public function register_actions(): void {
+		$counter = 1;
+
+		foreach ( self::$actions as $action ) {
+			try {
+				if ( ! as_has_scheduled_action( $action, [], $this->group ) ) {
+					// Schedule the action to run every day with a 5 second offset for each action.
+					as_schedule_recurring_action( time() + ( $counter * 5 ), DAY_IN_SECONDS, $action, [], $this->group );
+				}
+			} catch ( Exception $exception ) {
+				continue;
+			}
+
+			$counter++;
+		}
+	}
+
+	/**
+	 * Cancel all occurrence of scheduled action.
+	 *
+	 * @return void
+	 */
+	public function unregister_actions(): void {
+		foreach ( self::$actions as $action ) {
+			try {
+				if ( as_has_scheduled_action( $action, [], $this->group ) ) {
+					// Remove action from schedule.
+					as_unschedule_action( $action, [], $this->group );
+				}
+			} catch ( Exception $exception ) {
+				continue;
+			}
+		}
+	}
+
+	/**
+	 * Get AS actions.
+	 *
+	 * @return array Array of actions.
+	 */
+	public static function get_actions(): array {
+		return self::$actions;
+	}
+
+	/**
+	 * Register admin stat options.
+	 *
+	 * @return void
+	 */
+	public function register_stats_option() {
+		if ( false !== get_option( $this->stats_option ) ) {
+			return;
+		}
+
+		add_option( $this->stats_option, $this->admin_stats );
+	}
+
+	/**
+	 * Update stats option.
+	 *
+	 * @param string $key Option key.
+	 * @param mixed  $value Option value.
+	 * @return void
+	 */
+	public function update_stats( $key, $value ): void {
+		$stats = get_option( $this->stats_option );
+
+		if ( ! $stats ) {
+			$this->admin_stats[ $key ] = $value;
+			add_option( $this->stats_option, $this->admin_stats );
+
+			return;
+		}
+
+		$stats[ $key ] = $value;
+		update_option( $this->stats_option, $stats );
+	}
+
+	/**
+	 * Count number of attachments.
+	 *
+	 * @since  1.0
+	 * @author Jonathan Buttigieg
+	 *
+	 * @return void
+	 */
+	public function imagify_count_attachments(): void {
```

Comment:
You can remove the `imagify_` prefix from the functions you copied since they're in a class now